### PR TITLE
OCD-3586 - Update README documentation for each project

### PR DESCRIPTION
**Pull request by tmy1313**
OCD-3586

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,19 +17,38 @@ $ cd openchpl-sql
...
-changes in either a `chpl-data-model/OCD-XXXX.sql` file or the `chpl-data-model/v-next.sql` if the change will be deployed at the next deployment to production. The v-next.sql should be able to run multiple times without problems. If you are creating a table, `DROP TABLE IF EXISTS`. If you are dropping a table, `DROP TABLE IF EXISTS`. If you are altering a table, `DROP COLUMN IF EXISTS`. If data must be added or removed, ensure it is only added once, and removal removes only the correct data.
+### Updating Soft Deletes and Views
+To update the soft delete triggers, update the `dev/openchpl_soft-delete.sql` script to include updates.
+To update the views in the CHPL database, update the `dev/openchpl_views.sql` script to include updates.
 
-Every time a you pull down changes that affect the database run v-next.sql to get the latest database model. If you are running a specific OCD branch you might also need to run an OCD-XXXX.sql file to get the correct database.
+These scripts are run each time the database is updated.  These scripts will drop all database objects associatedw ith the script and recreate the objects.
```
Comment (andlar) [inline README.md]:
```suggestion
These scripts are run each time the database is updated. These scripts will drop all database objects associated with the script and recreate the objects.
```

File: README.md [hunk 2]
```diff
@@ -17,19 +17,38 @@ $ cd openchpl-sql
...
-If the changes made for development include the addition of a table, view, trigger, or any other thing that has ownership, check to see if you should have `--re-run grants \i dev/openchpl_grant-all.sql` at the end of the end of whatever script makes the changes.
+### Updating/Adding Tables and Data
+To update or add a new table to the CHPL database:
+* Create new file or append canges to existing `changes/ocd-XXXX.sql` which corresponds to the ticket the change is associated to
+* Add necessary `ALTER` statements, ensuring that the script can be run multiple times based on requirements
+  * `IF EXISTS`, `IF NOT EXISTS` clauses should be used to determine if the statement will be executed
 
-During a production push, the "ready to go live" `v-next.sql` file will be renamed to match the Version of the database that is going live, using an adaptation of the Semantic Versioning methodology. Later development will create a new `v-next.sql` file.
+To update data in tables
+* Create new file or append canges to existing `changes/ocd-XXXX.sql` which corresponds to the ticket the change is associated to
```
Comment (andlar) [inline README.md]:
```suggestion
* Create a new file or append changes to an existing `changes/ocd-XXXX.sql` file which corresponds to the ticket the change is associated to
```

File: README.md [hunk 3]
```diff
@@ -17,19 +17,38 @@ $ cd openchpl-sql
...
+* Add necessary `INSERT`, `UPDATE` statements, ensuring the script can be run multiple times based on requirements
+  * Using `WHERE NOT EXISTS` can often be used to help with determining if the statement should be executed
 
-## Data model load
+Sometimes database changes may require scripting not defined here to perform the required changes.  Those situations should be handled on a case by case basis.
+
+## Updating you local data to match another environment
```
Comment (andlar) [inline README.md]:
```suggestion
## Updating your local data to match another environment
```

File: README.md [hunk 4]
```diff
@@ -17,19 +17,38 @@ $ cd openchpl-sql
...
+  * `IF EXISTS`, `IF NOT EXISTS` clauses should be used to determine if the statement will be executed
 
-During a production push, the "ready to go live" `v-next.sql` file will be renamed to match the Version of the database that is going live, using an adaptation of the Semantic Versioning methodology. Later development will create a new `v-next.sql` file.
+To update data in tables
+* Create new file or append canges to existing `changes/ocd-XXXX.sql` which corresponds to the ticket the change is associated to
+* Add necessary `INSERT`, `UPDATE` statements, ensuring the script can be run multiple times based on requirements
+  * Using `WHERE NOT EXISTS` can often be used to help with determining if the statement should be executed
 
-## Data model load
+Sometimes database changes may require scripting not defined here to perform the required changes.  Those situations should be handled on a case by case basis.
```
Comment (andlar) [inline README.md]:
```suggestion
Sometimes database changes may require scripting not defined here to perform the required changes. Those situations should be handled on a case by case basis.
```

File: README.md [hunk 5]
```diff
@@ -17,19 +17,38 @@ $ cd openchpl-sql
...
+
+## Updating you local data to match another environment
+To update your local database to match a particular environment, pull the code associated with that environment.
+`git pull upstream/qa` for example.
 
+Run the `load-pending-changes.sh` script.  This will execute:
```
Comment (andlar) [inline README.md:37]:
```suggestion
Run the `load-pending-changes.sh` script. This will execute:
```

**Review by andlar — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -17,19 +9,38 @@ $ cd openchpl-sql
...
+To update the soft delete triggers, update the `dev/openchpl_soft-delete.sql` script to include updates.
+To update the views in the CHPL database, update the `dev/openchpl_views.sql` script to include updates.
 
-Every time a you pull down changes that affect the database run v-next.sql to get the latest database model. If you are running a specific OCD branch you might also need to run an OCD-XXXX.sql file to get the correct database.
+These scripts are run each time the database is updated.  These scripts will drop all database objects associated with the script and recreate the objects.
 
-If the changes made for development include the addition of a table, view, trigger, or any other thing that has ownership, check to see if you should have `--re-run grants \i dev/openchpl_grant-all.sql` at the end of the end of whatever script makes the changes.
+### Updating/Adding Tables and Data
+To update or add a new table to the CHPL database:
+* Create new file or append canges to existing `changes/ocd-XXXX.sql` which corresponds to the ticket the change is associated to
```
Comment (kekey1) [inline README.md]:
```suggestion
* Create new file or append changes to existing `changes/ocd-XXXX.sql` which corresponds to the ticket the change is associated to
```

File: README.md [hunk 7]
```diff
@@ -17,19 +9,38 @@ $ cd openchpl-sql
...
+Run the `load-pending-changes.sh` script.  This will execute:
+* All `ocd-XXXX.sql` scripts in the `/changes` directory
+* The `dev/openchpl_soft-delete.sql` script
+* The `dev/openchpl_views.sql` script
+* The `dev/openchpl_grant-all.sql` script (set permissions for all database objects)
+
+## Data model load
 Rename `dev/openchpl_role-template.sql` to `dev/openchpl_role.sql` and set the password for the `openchpl` role. These instructions assume the role/username used for the openchpl database is `openchpl`, and that the password in `openchpl_role.sql`, currently recorded as "change this password" will be update to match your installation. If the installer chooses to change the username/role, make sure it's also changed in the `openchpl.sql` file wherever the role is used.
 
 Next, run the script `dev/reset.sh` or `dev/reset.bat` from the `/dev` directory. These two scripts remove any previous OpenCHPL data model installation, with the associated roles, then recreate the required roles and databases, as well as fill out some of those database schemas with some required information.
```
Comment (kekey1) [inline README.md]: Edit to refer to the program as "CHPL" vs "OpenCHPL"?

**Review by kekey1 — CHANGES_REQUESTED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Review by palakpatel311 — APPROVED** [review]

**Pull request merged**